src/api.ts

```diff
@@ -120,6 +122,60 @@ export async function makeCoderSdk(baseUrl: string, token: string | undefined, s
   return restClient
 }
 
+/**
+ * Creates a fetch adapter using an Axios instance that returns streaming responses.
+ * This can be used with APIs that accept fetch-like interfaces.
+ */
+export function createStreamingFetchAdapter(axiosInstance: AxiosInstance) {
+  return async (url: string | URL, init?: FetchLikeInit) => {
+    const urlStr = url.toString()
+
+    const response = await axiosInstance.request({
+      url: urlStr,
+      headers: init?.headers as Record<string, string>,
+      responseType: "stream",
+      validateStatus: () => true, // Don't throw on any status code
+    })
+    const stream = new ReadableStream({
+      start(controller) {
+        response.data.on("data", (chunk: Buffer) => {
+          controller.enqueue(chunk)
+        })
+
+        response.data.on("end", () => {
+          controller.close()
+        })
+
+        response.data.on("error", (err: Error) => {
+          controller.error(err)
+        })
+      },
+
+      cancel() {
+        response.data.destroy()
+        return Promise.resolve()
+      },
+    })
+
+    const createReader = () => stream.getReader()
+
+    return {
+      body: {
+        getReader: () => createReader(),
```

Comment:
I think this could just be `getReader: stream.getReader`.

Reply:
Hmm, that seems to cause a compilation issue:

```
      TS2322: Type '(url: string | URL, init?: FetchLikeInit) => Promise<{ body: { getReader: ReadableStreamDefaultReader<any>; }; url: string; status: number; redirected: boolean; headers: { get: (name: string) => string | null; }; }>' is not assignable to type 'FetchLike'.
ts-loader-default_ab4862433a94faea
 @ ./src/extension.ts 47:29-60
```

But this compiles and seems to work:

```
getReader: () => stream.getReader(),
```

Updated.

Reply:
That's so weird I can't get `tsc` or VSCode to produce the error. What version of `tsc` do you have?

Reply:
Can't reproduce, I think I must have tried `getReader: createReader()` by accident. Updated with your simpler version.

Reply:
Nope, the simpler version doesn't work:
```
[2025-02-28T22:15:47.581Z] Value of "this" must be of type ReadableStream
```
Reverted back.

Reply:
So weird I can't get the error, even though the error message makes perfect sense. Just need to tweak my typescript setup I suppose.